From: Marc Zyngier <marc.zyngier@arm.com>
Date: Mon, 20 Mar 2017 17:18:06 +0000
Subject: arm64: cpu_errata: Add capability to advertise Cortex-A73 erratum
 858921
Git-commit: eeb1efbcb83c0cfe6d567abbacd675bbddf3d658
Patch-mainline: v4.12-rc1
References: fate#322150

In order to work around Cortex-A73 erratum 858921 in a subsequent
patch, add the required capability that advertise the erratum.

As the configuration option it depends on is not present yet,
this has no immediate effect.

Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
[mb: add define to cpufeatures.h instead of cpucaps.h]
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 Documentation/arm64/silicon-errata.txt |    1 +
 arch/arm64/include/asm/cpucaps.h       |    5 +++--
 arch/arm64/kernel/cpu_errata.c         |    8 ++++++++
 3 files changed, 12 insertions(+), 2 deletions(-)

--- a/Documentation/arm64/silicon-errata.txt
+++ b/Documentation/arm64/silicon-errata.txt
@@ -54,6 +54,7 @@ stable kernels.
 | ARM            | Cortex-A57      | #852523         | N/A                     |
 | ARM            | Cortex-A57      | #834220         | ARM64_ERRATUM_834220    |
 | ARM            | Cortex-A72      | #853709         | N/A                     |
+| ARM            | Cortex-A73      | #858921         | ARM64_ERRATUM_858921    |
 | ARM            | MMU-500         | #841119,#826419 | N/A                     |
 |                |                 |                 |                         |
 | Cavium         | ThunderX ITS    | #22375, #24313  | CAVIUM_ERRATUM_22375    |
--- a/arch/arm64/include/asm/cpucaps.h
+++ b/arch/arm64/include/asm/cpucaps.h
@@ -35,8 +35,9 @@
 #define ARM64_HYP_OFFSET_LOW			14
 /* #define ARM64_MISMATCHED_CACHE_LINE_SIZE	15 */
 #define ARM64_WORKAROUND_REPEAT_TLBI		16
-#define ARM64_HAS_NO_BCAST_TLBI                 17 /* Out of tree */
+#define ARM64_WORKAROUND_858921			17
+#define ARM64_HAS_NO_BCAST_TLBI                 18 /* Out of tree */
 
-#define ARM64_NCAPS				18
+#define ARM64_NCAPS				19
 
 #endif /* __ASM_CPUCAPS_H */
--- a/arch/arm64/kernel/cpu_errata.c
+++ b/arch/arm64/kernel/cpu_errata.c
@@ -132,6 +132,14 @@ const struct arm64_cpu_capabilities arm6
 			   MIDR_CPU_VAR_REV(0, 0)),
 	},
 #endif
+#ifdef CONFIG_ARM64_ERRATUM_858921
+	{
+	/* Cortex-A73 all versions */
+		.desc = "ARM erratum 858921",
+		.capability = ARM64_WORKAROUND_858921,
+		MIDR_ALL_VERSIONS(MIDR_CORTEX_A73),
+	},
+#endif
 	{
 	}
 };
